File: backend/app/models/responses/job.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel


class FileErrorResponse(BaseModel):
    upload_id: UUID
    filename: str
    stage: str
    message: str
    row_index: int | None


class JobErrorsResponse(BaseModel):
    files: list[FileErrorResponse]


class JobFileResponse(BaseModel):
    id: UUID
    job_id: UUID
    upload_id: UUID
    status: str
    rows_processed: int | None
    rows_failed: int | None
    error_detail: str | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime
    updated_at: datetime


class JobResponse(BaseModel):
    id: UUID
    project_id: UUID
    user_id: UUID
    schema_id: UUID
    status: str
    total_files: int
    completed_files: int
    failed_files: int
    errors: JobErrorsResponse | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime
    updated_at: datetime


class JobListResponse(BaseModel):
    items: list[JobResponse]
    next_cursor: str | None


class ExportResponse(BaseModel):
    id: UUID
    job_id: UUID
    user_id: UUID
    format: str
    size_bytes: int | None
    row_count: int | None
    status: str
    download_url: str | None
    expires_at: datetime | None
    created_at: datetime
    updated_at: datetime

File: frontend/types/job.ts
// Mirrors FastAPI JobResponse and CreateJobRequest models

export type JobStatus =
  | "queued"
  | "running"
  | "completed"
  | "completed_with_errors"
  | "failed";

export type FileError = {
  upload_id: string;
  filename: string;
  stage: string;
  message: string;
  row_index?: number;
};

export type Job = {
  id: string;
  project_id: string;
  user_id: string;
  schema_id: string;
  status: JobStatus;
  output_path: string | null;
  errors: { files: FileError[] } | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
};

export type CreateJobRequest = {
  project_id: string;
  schema_id: string;
  upload_ids: string[];
};
